src/robotFactory.js

```diff
@@ -1,10 +1,86 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(
+    name,
+    weight,
+    coords = {},
+    chipVersion,) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = coords;
```

Comment:
```suggestion
    this.coords = {
       x: coords.x || 0,
       y: coords.y || 0,
    };
```
Try this to make your code cleaner